src/components/ModalWithForm.js

```diff
@@ -0,0 +1,78 @@
+import Modal from "./Modal.js";
+
+export default class ModalWithForm extends Modal {
+  constructor({ modalSelector, onSubmitCallback }) {
+    super({ modalSelector });
+
+    //note: onSubmitCallback must always return a promise.
+    this._onSubmitCallback = onSubmitCallback;
+    this._submitButton = this._modalElement.querySelector(
+      ".modal__submit-button"
+    );
+    this._inputFields = this._modalElement.querySelectorAll(".modal__input");
+    this._form = this._modalElement.querySelector(".modal__form");
+    this._currentProfileName = document.querySelector(".profile__name");
+    this._currentProfileDescription = document.querySelector(
+      ".profile__description"
+    );
+  }
+
+  getFormElement() {
+    return this._form;
+  }
+
+  _resetInputs() {
+    this._inputFields.forEach((field) => {
+      field.value = "";
```

Comment:
You can easily clear the inputs with `reset`. Every **form** has this method

```js

  _resetInputs() {
    this._form.reset()
  }
```

Reply:
Of course this already exists. I should have remembered. Thank you!